cmd/thanos/main.go

```diff
@@ -295,21 +296,25 @@ func defaultGRPCServerOpts(logger log.Logger, reg *prometheus.Registry, tracer o
 }
 
 // metricHTTPListenGroup is a run.Group that servers HTTP endpoint with only Prometheus metrics.
-func metricHTTPListenGroup(g *run.Group, logger log.Logger, reg *prometheus.Registry, httpBindAddr string) error {
+func metricHTTPListenGroup(g *run.Group, logger log.Logger, reg *prometheus.Registry, httpBindAddr string, component component.Component) (*prober.Prober, error) {
```

Comment:
We need to work on name of  this if we really want this API ` (*prober.Prober, error)` - current state is really confusing. Name of function/method should explicitly describe what is done. I think because we merged prober creation with HTTP server init, it will be actually hard to find meaningful name so maybe it's fine to keep it separated? not sure.

Reply:
Yes.. the name is misleading you are right. I didn't think about it since it was there before. 

Splitting it sounds reasonable to me. Will do

Reply:
Not resolved at all

Reply:
I renamed it to `defaultHTTPListener` more info in comment https://github.com/improbable-eng/thanos/pull/656#discussion_r277271219